src/commands/durableTaskScheduler/deleteTaskHub.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import {type IActionContext } from "@microsoft/vscode-azext-utils";
+import { type DurableTaskSchedulerClient } from "../../tree/durableTaskScheduler/DurableTaskSchedulerClient";
+import { localize } from "../../localize";
+import { type DurableTaskHubResourceModel } from "../../tree/durableTaskScheduler/DurableTaskHubResourceModel";
+import { type MessageItem } from "vscode";
+import { withAzureActivity } from "../../utils/AzureActivity";
+import { withCancellation } from "../../utils/cancellation";
+
+export function deleteTaskHubCommandFactory(schedulerClient: DurableTaskSchedulerClient) {
+    return async (actionContext: IActionContext, taskHub: DurableTaskHubResourceModel | undefined): Promise<void> => {
+        if (!taskHub) {
+            throw new Error(localize('noTaskHubSelectedErrorMessage', 'No task hub was selected.'));
```

Comment:
Should we hide some of these commands from the command palette so this error never happens?

Reply:
That's a good idea, updated.  I *do* intend to eventually support a more palette-driven workflow, but not for this iteration.